### DISCUSSION TITLE
Empty commits and `paths`/`paths-ignore`

### DISCUSSION THREAD

**Discussion by ghost** [community/community#9182] [Actions]
Hello, I have a workflow which triggers on a `push`/`pull_request` event and I wish to:
1) ignore certain files/folders with `paths`/`paths-ignore`,
2) but still run the workflow where there are no changes.

This is impossible at the moment. To reproduce the issue, create a workflow A with the following trigger:

```yaml
on: push
```

And a workflow B with:

```yaml
on:
  push:
    paths-ignore:
      - 'doc/**'
```

Then push an empty commit:

```
git commit --allow-empty -m "empty commit"
git push
```

Workflow A will trigger but not workflow B.

[The documentation](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#git-diff-comparisons) states:

> The filter determines if a workflow should run by evaluating the changed files and running them against the paths-ignore or paths list. If there are no files changed, the workflow will not run.

Which makes it sound like it's an expected behavior, but it could also be a failure to account for a somewhat peculiar case. It seems the only workaround is to implement a filter in a job or step, which obviously is not great.

Related: https://github.community/t/paths-ignore-option-causes-empty-amend-commits-to-be-ignored/18232

**Comment by HariSekhon** [comment]
Why not just use `workflow_dispatch` when you want to trigger a workflow?

This manual triggering can also be done via GitHub CLI.

You don't have to go around adding empty commits to your repo just to trigger a workflow.

The only problem is that `workflow_dispatch` trigger doesn't work until the workflow its in has been merged to the default branch, so the gap here is where you've added a new workflow in the your branch / PR and then want to re-trigger as part of testing before merging the PR, although that is implicitly not the case with your `on: [push]`.

**Comment by paescuj** [comment]
You may want to take a look at https://github.com/fkirc/skip-duplicate-actions.

**Comment by surgiie** [comment]
Kinda wish you could force run workflows and tell github actions to ignore paths somehow, something like:

`git commit --allow-empty -m "[force-ci]"`

Similar to how github lets you skip workflows: 

https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs

**Comment by na-jakobs** [comment]
We are also seeing this issue, this is a bug 🐛 in GitHub.

When empty commits are pushed, normally workflows are triggered, but not when a workflow has `paths-ignore`.

**Comment by myrzan** [comment]
It's a pretty useless feature if it doesn't trigger a new build when creating a new branch, even if there are no changes. Either this behavior needs to be fixed, which I imagine would cost several LLM minutes, or I don't know.
